File: src/MoneyManager.Domain/Entities/Budget.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace MoneyManager.Domain.Entities;

public class Budget
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    [BsonElement("userId")]
    public string UserId { get; set; } = string.Empty;

    [BsonElement("month")]
    public string Month { get; set; } = string.Empty; // YYYY-MM format

    [BsonElement("items")]
    public List<BudgetItem> Items { get; set; } = [];

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

[BsonIgnoreExtraElements]
public class BudgetItem
{
    [BsonElement("categoryId")]
    public string CategoryId { get; set; } = string.Empty;

    [BsonElement("limitAmount")]
    public decimal LimitAmount { get; set; }

    [BsonElement("spentAmount")]
    public decimal SpentAmount { get; set; }
}

File: src/MoneyManager.Frontend/src/types/budget.ts
export interface BudgetItemDto {
  categoryId: string;
  limitAmount: number;
}

export interface BudgetRequestDto {
  month: string;
  items: BudgetItemDto[];
}

export interface BudgetItemResponseDto {
  categoryId: string;
  categoryName: string;
  categoryColor: string;
  limitAmount: number;
  spentAmount: number;
}

export interface BudgetResponseDto {
  id: string;
  month: string;
  items: BudgetItemResponseDto[];
}
